src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -286,7 +305,14 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
 
                 if (_options.OfflineCache != null && cachedData == null)
                 {
-                    _options.OfflineCache.Export(_options, JsonSerializer.Serialize(data));
+                    IDictionary<string, ConfigurationSetting> offlineCacheData = new Dictionary<string, ConfigurationSetting>();
+
+                    foreach (KeyValuePair<string, List<ConfigurationSetting>> setting in data)
+                    {
+                        offlineCacheData[setting.Key] = setting.Value.Last();

```

Comment:
>offlineCacheData[setting.Key] = setting.Value.Last(); [](start = 24, length = 53)

Do we need to create another dictionary and copy the last values, or could we re-use the current dictionary and serialize only the last values?

Reply:
Fixed